src/helpers.js

```diff
@@ -17,30 +17,39 @@ const distinct = (value, index, self) => {
   return self.indexOf(value) === index
 }
 
-const directoryExists = directory => {
-  return (
-    directory &&
-    fs.existsSync(directory) &&
-    fs.lstatSync(directory).isDirectory()
-  )
+const directoryExists = async directory => {
+  try {
+    return (await fsPromises.lstat(directory)).isDirectory()
+  } catch (err) {
+    // this will happen if we don't have access to the directory or if it doesn't exist
+    return false
+  }
 }
 
-const createOrOverwriteFile = (file, body) => {
-  const dirName = path.dirname(file)
-  if (!directoryExists(dirName)) {
-    fs.mkdirSync(dirName, { recursive: true })
-  } else if (fs.existsSync(file)) {
-    fs.unlinkSync(file)
+const fileExists = async path => {
+  try {
+    return (await fsPromises.lstat(path)).isFile()
+  } catch (err) {
+    // this will happen if we don't have access to the file or if it doesn't exist
+    return false
   }
-  fs.writeFileSync(file, body)
 }
 
-const findDepartmentByNumber = departmentNumber => {
-  return departmentsJson.find(department => {
-    return department["depNo"] === departmentNumber.toString()
-  })
+const createOrOverwriteFile = async (file, body) => {
+  const dirName = path.dirname(file)
+  if (!(await directoryExists(dirName))) {
+    await fsPromises.mkdir(dirName, { recursive: true })
+  } else if (await fileExists(file)) {
+    await fsPromises.unlink(file)
```

Comment:
When running the conversion subsequent times after the first with the `--download` flag turned on, I get an error from this line:

```sh
Error: Error: Error: ENOENT: no such file or directory, '
    at Object.unlink (/Users/c4103/Github/ocw-to-hugo/src/fsPromises.js:8:13)
    at createOrOverwriteFile (/Users/c4103/Github/ocw-to-hugo/src/helpers.js:43:22)
    at async writeS3Object (/Users/c4103/Github/ocw-to-hugo/src/aws_sync.js:88:5)
    at async Promise.all (index 3)
    at async downloadCourseRecursive (/Users/c4103/Github/ocw-to-hugo/src/aws_sync.js:92:3)
    at async /Users/c4103/Github/ocw-to-hugo/src/aws_sync.js:56:7
    at async Promise.all (index 24)
    at async downloadCourses (/Users/c4103/Github/ocw-to-hugo/src/aws_sync.js:50:10)
    at async run (/Users/c4103/Github/ocw-to-hugo/src/bin/index.js:53:5) {
  message: [Error: ENOENT: no such file or directory, unlink 'private/courses/6-033-computer-system-engineering-spring-{
    errno: -2,
    code: 'ENOENT',
    syscall: 'unlink',
    path: 'private/courses/6-033-computer-system-engineering-spring-2018/0361d9ecba6ae41417b214597ecbe5ac_r2_-2KW76ec.s'
  }
}
```

Reply:
Thanks, I'll investigate. I also found a different bug I'm investigating.

Reply:
I added the `aws_sync` class so we could maybe have finer control over what gets synced that you can achieve with arguments into `aws s3 sync` but maybe at some point it's worth discussing just using `aws s3 sync` instead to avoid extra issues.  It is convenient to be able to just use the `--download` flag, there are benefits and tradeoffs.

Reply:
@gumaerc Can you try this again and see if you still see the error?

Reply:
all fixed :+1: